{{ cookiecutter.github_repo_name }}/src/{{ cookiecutter.package_dir_name }}/calculator.py

```diff
@@ -2,5 +2,42 @@
 
 
 def dot_product(a, b):
-    """Calculate the dot product of two vectors."""
-    return np.dot(a, b)
+    """Compute the dot product of two vectors of any size.
+
+    Ensure that the inputs of a and b are of the same size.
+    The supported types are Python list, Python tuple, and NumyPy array.
+
+    Parameters
+    ----------
+    a : array_like
+        The first input of vector.
+    b : array_like
+        The second input vector.
+
+    Returns
+    -------
+    float
+        The dot product of the two vectors.
+
+    Examples
+    --------
+
+    Compute the dot product of two lists:
+    >>> a = [1, 2, 3]
+    >>> b = [4, 5, 6]
+    >>> dot_product(a, b)
+    32.0
+
+    Compute the dot product of two tuples:
```

Comment:
please delete below here.   We don't need to give examples where just different types are being given to the function.

Reply:
fixed